src/components/bookDetails/BookDetails.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from 'react'
+
+import { useNavigate, useParams } from 'react-router-dom'
+import { TfiBackLeft as IconForBack } from 'react-icons/tfi'
+
+import './BookDetails.css'
+import placeholderImg from '../../assets/placeholder/placeholderForBook.png'
+import { BookDetailsRequest } from '../../model/BookRequest'
+import BookService from '../../services/BookService'
+import ModalCreateBooks from '../modal/ModalCreateBooks'
+import ModalForEdit from '../modalForEdit/ModalForEdit'
+import { convertDateToString, s } from '../../utils/ConvertDate'
+
+
+
+
+const BookDetails = () => {
+  const { id } = useParams()
+  const navigate = useNavigate()
+  const [ isOpenEditModal, setIsOpenEditModal ] = useState(false)
+  const [ bookDetails, setBookDetails ] = useState<BookDetailsRequest>(
+    {
+      Id: 0,
+      Title: '',
+      Authors: [],
+      Cover: '',
+      Available: 0,
+      Description: '',
+      ISBN: '',
+      PublishDate: '',
+      Quantity: 0
+    }
+  )
+
+  useEffect(() => {
+    getBook()
+  }, [ ])
+
+  const getBook = () => {
+    if(!id) return
+    BookService.getBook(parseInt(id)).then((response) => {
+      setBookDetails(response.data)
+    }).catch((error) => console.error(error))
+  }
+
+  const handleBack = () => {
+    navigate('/mainpage')
+  }
+
+  const handleDeleteBook = () => {
+    BookService.deleteBook(bookDetails.Id)
+      .then(() => {
+        navigate('/mainpage')
```

Comment:
Does this work as expected? You navigate to main page first and then confirm?

Reply:
No... I try fix this